### DISCUSSION TITLE
Remove code annotations from PDF output

### DISCUSSION THREAD

**Discussion by DillonHammill** [quarto-dev/quarto-cli#6316] [Q&A] [answered]
### Description

I would like to display code annotations with hover in rendered HTML format but NOT in PDF format. I have included the following options in my `_quarto.yml`:

```{r}
format:
  html:
    theme: cosmo
    code-annotations: hover
  pdf:
    documentclass: scrreprt
    header-includes:
      - \usepackage{setspace}\doublespacing
      - \usepackage{ragged2e}
    code-annotations: false
```

I am able to successfully render HTML with code annotations but I encounter errors when trying to render the PDF without code annotations:
```{r}
Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999994 (TeX Live 2022) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages
ERROR: 
compilation failed- error
Undefined control sequence.
<argument> \circled 
                    {1}
l.5010 ... }\hspace*{\fill}\NormalTok{\circled{1}}
```

Also is there any plan to treat code annotations as standard text that can be arranged with code/figures (e.g. on the right of a code block)?

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/authoring/code-annotation.html>.

**Reply by cderv** [reply]
@mcanouil what are you pointing to exactly ? It seems to me that `code-annotations: false` under `pdf` format, should indeed not process the annotation and add a LaTeX code undesired. 

It seems to me like a bug where we don't take into account `code-annotations: false`

**Reply by mcanouil** [reply]
There are two different things in the post and no code example.

The issue with LaTeX (bug, see below) and

> Also is there any plan to treat code annotations as standard text that can be arranged with code/figures

Which means `code-annotations: below` or `code-annotations: false` but since the desired output is unclear and the lack of code, hard to tell.

---

Bug with `code-annotations: false`.

````qmd
---
format: pdf
---

```r
1 + 1 # <1>
```

1. Simple addition.
````

Quarto still adds `\hspace*{\fill}\NormalTok{\circled{1}}` which when `code-annotations: false` is not defined.

Which can be "fixed" (but actually the code cell should be left untouched in this case) with the following:

````qmd
---
format: pdf
code-annotations: false
include-in-header:
  - text: |
      \newcommand*\circled[1]{\tikz[baseline=(char.base)]{\node[shape=circle,draw,inner sep=1pt] (char) {{\scriptsize#1}};}} 
---

```r
1 + 1 # <1>
```

1. Simple addition.
````

**Reply by cderv** [reply]
I have opened #6320 to track the issue with `code-annotations: false` and LaTeX output

**Comment by mcanouil** [comment, marked as answer]
To note and to be clear:
- "without annotation" means removing annotations -> `code-annotations: none`, see <https://quarto.org/docs/authoring/code-annotation.html#removing-annotations>.
- with annotation but as standard text (comment remains a comment in the code cell and text is a plain LaTeX list) -> `code-annotations: false`[^1], see <https://quarto.org/docs/authoring/code-annotation.html#disabling-annotation>.

[^1]: Currently not working, see <https://github.com/quarto-dev/quarto-cli/issues/6320>.

**Reply by cderv** [reply]
> "without annotation" means removing annotations -> code-annotations: none, see [quarto.org/docs/authoring/code-annotation.html#removing-annotations](https://quarto.org/docs/authoring/code-annotation.html#removing-annotations).

In fact we may need then to fix this in #6320 too. Because currently `none` also add the `circled` command, and  it is shown in output. 
Only the comment detail below cell is removed. I understand it should not be the case from 
> you can set code-annotations: none, which will remove the annotation comments from your code and suppress the output of the ordered list which contains the annotation text.

**Reply by mcanouil** [reply]
From the documentation, the circle should not be added.
Worth checking for other code annotations compatible formats.

**Reply by DillonHammill** [reply]
Thanks for the clarification @mcanouil, I can confirm that I am able to render the PDF with `code-annotations: none`. However, as @cderv pointed out the numbered circles still appear in the code chunk with strange alignment. It would be great if we could remove them altogether when `code-annotations: none`.
![image](https://github.com/quarto-dev/quarto-cli/assets/28646736/4d4be244-43ae-444b-beec-c1cc149b91f4)

**Reply by cderv** [reply]
Yes @DillonHammill issue is tracked in https://github.com/quarto-dev/quarto-cli/issues/6332 for `code-annotations: none`

**Reply by cderv** [reply]
@DillonHammill every fix is now merged. Hopefully with next pre-release, it will all work better!
